### PR TITLE
Add EUDEMO IVC builder

### PR DESCRIPTION
**Pull request by hsaunders1904**
# Work in progress

## Description

Adds a variable thickness blanket to the first wall builder.

I've messed up a rebase somehow, so there are lots of extra commits in this PR, but the files changed are correct. I'll fix the commits issue

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/builders/EUDEMO/reactor.py [hunk 1]
```diff
@@ -63,7 +64,6 @@ def run(self) -> Component:
         component.add_child(self.build_plasma())
         component.add_child(self.build_TF_coils(component))
         component.add_child(self.build_PF_coils(component))
-        component.add_child(self.build_first_wall(component))
```
Comment (hsaunders1904) [inline bluemira/builders/EUDEMO/reactor.py:66]: @CoronelBuendia I pictured calling `build_in_vessel_component_shapes()` here, then using those shapes to call whichever other builders need them. If you had something else in mind, feel free to go with that.

The `build_ivc_xz_shapes()` function will turn the output components of the `IVCBuilder` into the shapes you need

File: bluemira/builders/EUDEMO/ivc/ivc.py [hunk 2]
```diff
@@ -23,18 +23,20 @@
...
 
 import numpy as np
 
 from bluemira.base.builder import BuildConfig, Component
-from bluemira.base.components import PhysicalComponent
+from bluemira.base.components import Component, PhysicalComponent
 from bluemira.base.error import BuilderError
-from bluemira.builders.EUDEMO.first_wall.divertor import DivertorBuilder
-from bluemira.builders.EUDEMO.first_wall.wall import WallBuilder
+from bluemira.builders.EUDEMO.ivc.blanket import BlanketBuilder
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/ivc/ivc.py:33]: Maybe `IVCThicknessBuilder` or `BlanketThicknessBuilder`?

File: bluemira/builders/EUDEMO/ivc/ivc.py [hunk 3]
```diff
@@ -23,18 +23,20 @@
...
+from bluemira.builders.EUDEMO.ivc.divertor import DivertorBuilder
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/ivc/ivc.py:34]: `DivertorSilhouetteBuilder`?

File: bluemira/builders/EUDEMO/reactor.py [hunk 4]
```diff
@@ -282,14 +282,16 @@ def build_first_wall(self, component_tree: Component, **kwargs):
...
             self._params.to_dict(), build_config=config, equilibrium=plasma.equilibrium
         )
         self.register_builder(builder, name)
 
         component = super()._build_stage(name)
         bluemira_print(f"Completed design stage: {name}")
-        return component
+        # TODO(hsaunders1904): this is not functional at the moment,
+        # need to check parameter access is correct
+        return build_ivc_xz_shapes(component, self._params.get_param("c_rm").value)
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/reactor.py:294]: I normally do `self._params.c_rm.value` not sure if that would work at the reactor level

File: bluemira/builders/EUDEMO/ivc/blanket.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        Build the component.
+        """
+        blanket = Component(self.name)
+        blanket.add_child(self._build_xz())
+        return blanket
+
+    def _build_xz(self) -> Component:
+        """Build the components in the xz-plane."""
+        # TODO(hsaunders): add thickness parameters to input params dict
+        boundary_wire = varied_offset(self._wall_shape, 1, 2.5, 45, 175)
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/ivc/blanket.py:71]:
```suggestion
        boundary_wire = varied_offset(self._wall_shape, self._params.tk_bb_ib.value, self._params.tk_bb_ob.value, 60, 160)
```

File: bluemira/builders/EUDEMO/ivc/blanket.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    .. code-block::
+
+        blanket (Component)
+        └── xz (Component)
+            └── blanket_boundary (PhysicalComponent)
+    """
+
+    COMPONENT_BOUNDARY = "blanket_boundary"
+
```
Comment (CoronelBuendia) [inline bluemira/builders/EUDEMO/ivc/blanket.py:42]: Don't we need `_required_params = [..]` here?

**Review by CoronelBuendia — CHANGES_REQUESTED** [review]

**Pull request merged**